src/tools/pub/glossary.md

```diff
@@ -29,6 +29,74 @@ specify `any` for their dependencies' [version constraints][].
 [lockfiles]: #lockfile
 [version constraints]: #version-constraint
 
+## Content hashes
+
+The pub.dev repository serves a sha256 hash of each package version it hosts.
+The pub client can use this hash to validate the integrity of downloaded packages,
+and protect against changes on the repository. 
+
+When `dart pub get` downloads a package, it computes the hash of the
+downloaded archive. The hash of each hosted dependency is stored with the
+[resolution][] in the [lockfile][].
+
+The pub client uses this content hash to verify that running `pub get` again using
+the same lockfile, potentially on a different computer, uses exactly the same packages.
+
+If the locked hash doesn't match what's currently in the pub cache,
+pub redownloads the archive. If it still doesn't match, the lockfile
+updates and a warning is printed. For example:
+
+```
+$ dart pub get
+Resolving dependencies...
+Cached version of foo-1.0.0 has wrong hash - redownloading.
+ ~ foo 1.0.0 (was 1.0.0)
+The existing content-hash from pubspec.lock doesn't match contents for:
+ * foo-1.0.0 from "pub.dev"
+This indicates one of:
+ * The content has changed on the server since you created the pubspec.lock.
+ * The pubspec.lock has been corrupted.
+ 
+The content-hashes in pubspec.lock has been updated.
+
+For more information see:
+https://dart.dev/go/content-hashes
+
+Changed 1 dependency!
+```
+
+The updated content hash will show up in your version control diff,
+and should make you suspicious.
+
+To make a discrepancy become an error instead of a warning, use
+[`dart pub get --enforce-lockfile`][]. It will cause the resolution to fail
+if it cannot find package archives with the same hashes, without updating the lockfile.
```

Comment:
**issue:** Include some version of this in the explanation of `--enforce-lockfile`.

Reply:
Which part exactly? It seems to me to say the exact same thing: "...will refuse to resolve
if anything deviates from `pubspec.lock`"

Reply:
This copy block provides context to resolution: package hashes don't match and you can't update the lock file. This is a great explanation.